pkg/install/v1alpha1/cstor_pool.go

```diff
@@ -159,10 +159,12 @@ spec:
     status:
       phase: Init
     versionDetails:
-      current: {{ .CAST.version }}
+      status:
+        current: {{ .CAST.version }}
+        dependentsUpgraded: true
+        state: RECONCILED
```

Comment:
do we need to consider setting the `state` as well?
If so, lot of changes might be required, like, setting to 'Reconciling' in upgrade code, setting to 'error' or 'reconciled' in volumeReconciler functions.

Reply:
its good to get the `state` as well updated.. but, it involves lot of changes
for now, we should check for 'current' and 'desired' versions to see whether upgrade is done or not..
Updating 'reason' or 'message' should be fine

Reply:
So in the job instead of watching the state the job will look for non empty reason and message. Correct ?

Reply:
yes.. thats what I guess, unless you think you can take care of `state` as well